File: iOS/TUIKit/TUICallKit/TUICallKit-Swift/State/TUICallEvent.swift
//
//  TUICallEvent.swift
//  TUICallKit
//
//  Created by vincepzhang on 2023/4/3.
//

let EVENT_KEY_USER_ID = "userId"
let EVENT_KEY_CODE = "code"
let EVENT_KEY_MESSAGE = "message"

public class TUICallEvent {
    enum EventType {
        case ERROR
        case TIP
        case UNKNOWN
    }

    enum Event {
        case USER_REJECT
        case USER_NO_RESPONSE
        case USER_LINE_BUSY
        case USER_LEAVE
        case USER_EXCEED_LIMIT
        
        case ERROR_COMMON
        case UNKNOWN
    }
    
    var eventType: EventType
    var event: Event
    var param: [String: Any]

    init(eventType: EventType, event: Event, param: [String: Any]) {
        self.eventType = eventType
        self.event = event
        self.param = param
    }
}

File: Android/TUIKit/TUICallKit/tuicallkit-kt/src/main/java/com/tencent/qcloud/tuikit/tuicallkit/state/TUICallEvent.kt
package com.tencent.qcloud.tuikit.tuicallkit.state

public class TUICallEvent {
    public var eventType: EventType? = null
    public var event: Event? = null
    public var param: HashMap<String, Any?>? = null

    constructor(eventType: EventType, event: Event, param: HashMap<String, Any?>?) {
        this.eventType = eventType
        this.event = event
        this.param = param
    }

    enum class EventType { ERROR, TIP }

    enum class Event {
        USER_REJECT,
        USER_NO_RESPONSE,
        USER_LINE_BUSY,
        USER_LEAVE,
        USER_EXCEED_LIMIT,

        ERROR_COMMON,
    }

    companion object {
        const val EVENT_KEY_USER_ID = "userId"
        const val EVENT_KEY_CODE = "code"
        const val EVENT_KEY_MESSAGE = "message"
    }
}